### ISSUE TITLE
Display error in Output window from interpreter message with non-ASCII character

### ISSUE THREAD

**Issue by robertlzj** [pkulchenko/ZeroBraneStudio#663]
Output window can't display correctly on non-ASCII characters in message from interpreter, like error message, or path in stack traceback. So it's hard to debug without a clue.

I found interpreter like `lua.exe` in `ZeroBraneStudio\bin` could work fine with correct display about these message.
BTW, Interpreter will give prompt with some part in localized language..?

**Comment by pkulchenko** [comment]
@robertlzj, do you have a specific example with the characters you are attempting to show and how they look like in the Output window? The current implementation should handle utf-8 valid characters without issues, but will replace invalid characters with `[SYN]` markers; maybe that's what you are referring to. The wxwidgets31x-upgrade branch code will show those invalid characters as-is, so after the changes in in that branch (as discussed in #260) are merged, this issue should be resolved. If that's the case, I'm closing this one, but if it's something else, please reopen with more information and examples.

**Comment by robertlzj** [comment]
- In Output window, the file name, error message is not identifiable, the long path is also truncated. 
![image](https://cloud.githubusercontent.com/assets/7419030/16624053/d48165b4-43d2-11e6-94fc-40131906a38b.png)
- In `lua.exe`, using same path & file. Error message and path in stack traceback could display correctly.
![image](https://cloud.githubusercontent.com/assets/7419030/16624108/0e2db8a8-43d3-11e6-923a-ce94b6ea2151.png)
Additional information:
- The content of lua document is `require 'iuplua'`
- The chinese content of error message means "Can't find the module" (the module refer to `iup.dll` which should be loading by `iuplua.dll` which is an agent/wrapper I think)

**Comment by pkulchenko** [comment]
@robertlzj, can you try to run the same thing from the wxwidgets31x-upgrade branch? I'd be interested in how it's handled there.

**Comment by pkulchenko** [comment]
@robertlzj, I tried with different Unicode characters (russian) and get the expected result:

This is during debugging:

```
привет/файл.lua:2: attempt to call global 'print1' (a nil value)
stack traceback:
	привет/файл.lua: in main chunk
```

I won't be able to reproduce the same error message, but I can at least try with the same file names if you copy them as text instead of the image. What is the OS version and the file system type?

**Comment by robertlzj** [comment]
The wxwidgets31x-upgrade branch didn't help, `[SYN]` only change to hex.
The problem seems just the part returned from interpreter.
Use your test file name.
![zeroBrane output window](http://i4.piimg.com/567571/6170e111a8e938e1.jpg)
The result directly from lua.exe is correct.
![interpreter window](http://i4.piimg.com/567571/7f63ad506d9113d1.png)

Full path of file`D:\程序\Lua\файл.lua`, content is `print1()`.
OS: Win 10 64bit NTFS

**Comment by pkulchenko** [comment]
@robertlzj, this gets even more interesting; the message you have at the top of the output ("Can't get short path for a Unicode file name...") is only shown when the Lua interpreter can't open the file, but the file exists in the file system, so I'm not sure how it's being opened. On my system when I try to run it on the file like `程序\файл.lua` and disable conversion to the short names, I get an error: `ZeroBraneStudio\bin\lua.exe: cannot open ...\??\????.lua: Invalid argument`.

When I try to debug the same file, I get the proper output, but this is because debugging doesn't launch the interpreter with the Unicode file name as a parameter:

```
程序/файл.lua:2: attempt to call global 'print1' (a nil value)
stack traceback:
	程序/файл.lua: in main chunk
```

It seems like the issue is caused by launching the interpreter with a Unicode name when the file system doesn't support the conversion to the "short" file names. It works in the second example you have only because you use `dofile`; try running the lua interpreter and passing that Unicode file name as the parameter and look at the error message.

**Comment by robertlzj** [comment]
- With your promt

    >  .. is only shown when the Lua interpreter can't open the file

    I test and find can't open file in Unicode name by double-click (in test before, I creat these special named file in ZB), shows file name in hex.
    ![Can't open file in Unicode name by double-click](http://i4.piimg.com/567571/5b6eb6a475a26c68.png)
    Can open it by drag & drop.
- running the lua interpreter and passing that Unicode file name as the parameter
![running the lua interpreter and passing that Unicode file name as the parameter](http://i2.piimg.com/567571/e8e30f14a940a0e9.png)

- Associate with the problem ZB can't open file in Unicode name, I'm worry, there may be something wrong with my file system!? I can't upload pictures using github's comment function these days since the the first replay with picture.. 
But the file name of picture is ASCII, and this occur after the ZB problem. 
I have to check this out. Any suggestion? I tried `chkdsk` already.

**Comment by pkulchenko** [comment]
@robertlzj, I think your file system is fine, but the settings seem to be different from my local settings.

What version of Windows are you running? What is the file system on the disk you are running the IDE from? Can you show the output of `chcp` command? Also, if you have `locale` command installed, can you include the output of `locale` command? Thanks.

**Comment by robertlzj** [comment]
`chcp` is 936. 
There is no `locale` command on my machine, I think you mean locale **ZH-CN**?
OS: win10 home edition.
File system: NTFS on C, exFAT on D (SD card), on tablet computer.
I used to place ZB and/or Lua document on D, but I did'nt find relevance. 
Although strange thing occured. Didn't save the picture at that time, I annotated bellow
![](http://i4.piimg.com/567571/beb1c40341724b5c.png)
The red line occured to be `D:\程序\Lua` which used to place Lua document (, the config of ZB?), could see this path in screenshot in previous reply, BUT, within the test, both ZB (including Lua.exe) and Lua document are on C:. So, I didn't know why there appeared the isolated path.

**Comment by robertlzj** [comment]
Other test, using ZB on disk D with Unicode characters in path, opening a normal file.
![](http://i1.piimg.com/567571/5c5c49fc8dd1bad3.png)
The yellow folder name should be the same as the red one.
I didn't know what happened there.
Did these two ZB copy conflict? (I should have delete the one on D...)

**Comment by pkulchenko** [comment]
@robertlzj, ok, can you run `msinfo32` and copy the value of `Locale` item on the list?

I thought that the issue only applies to the Unicode file names, but according to your first message, it may apply to a text too.

Let's go back to the beginning. Can't you add `print("程序 файл")` to your script and execute it? I'd like to see how this text is represented in the Output panel.

**Comment by pkulchenko** [comment]
@robertlzj, thank you for the additional output; interesting results.

Can you show me two more values: `wx.wxStandardPaths.Get():GetExecutablePath()` and `ide.editorFilename`? You can simply type it into the Local console and post the output here.

**Comment by pkulchenko** [comment]
> Did these two ZB copy conflict? (I should have delete the one on D...)

No, the two instances should not conflict, but they may theoretically show some different results depending on what the local filesystem configuration is, so I'd be interested in the results of the two commands in the previous comment that you can run from the IDE installed on disk `C` and disk `D`.

**Comment by robertlzj** [comment]
- The value of `locale` (if I understand this word) is localized text, that is `the People's Republic of China ; (PRC)` in English.
- `print("程序 файл")  ` execute and output correct as it is - `程序 файл`. (The problem occured in message from interpreter when code error?)
- The two values are:
  - Using ZB on disk C: `C:\\ProgramFiles\\ZeroBraneStudio\\bin\\wx.dll` and `C:\\ProgramFiles\\ZeroBraneStudio\\zbstudio.exe`
  - Using ZB on disk D: `D:\\软件\\ZeroBraneStudio-master\\bin\\wx.dll` and `D:\\软件\\ZeroBraneStudio-master/D:\\\200\237\188\254\\ZeroBraneStudio-master\\zbstudio.exe`  
  I seems had used ZB on disk D before, no problem yet.

**Comment by robertlzj** [comment]
Hi~
Problem still exist on my new computer / new system, also latest ZeroBrane with default setting.
The problem is: in Output window, the trace information from interpreter shows abnormal. concretely, characters which are not ASCII, shows uncorrectly - multibyte character shows their  hexademical code, and long string path truncated to short version.
On Cmd / Lua interpreter window, these trace information shows correctly.

Windows 10 64-bit 1703

**Comment by pkulchenko** [comment]
This should be resolved with the fix applied for #804.
